clients/go/admin/token_source.go

```diff
@@ -0,0 +1,42 @@
+package admin
+
+import "golang.org/x/oauth2"
+import "context"
+
+// This class is here because we cannot use the normal "github.com/grpc/grpc-go/credentials/oauth" package to satisfy
+// the credentials.PerRPCCredentials interface. This is because we want to be able to support a different 'header'
+// when passing the token in the gRPC call's metadata. The default is filled in in the constructor if none is supplied.
+type TokenSourcePerCallCredential struct {
```

Comment:
Is TokenSourcePerCallCredential the final name?

Reply:
unless you tell me what to replace it with.

Reply:
I'll defer to @EngHabu